lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -3402,6 +3402,26 @@ class SPIRVJointMatrixINTELWorkItemInst : public SPIRVJointMatrixINTELInstBase {
 _SPIRV_OP(JointMatrixGetElementCoord, true, 5)
 #undef _SPIRV_OP
 
+class SPIRVCooperativeMatrixInvocationInstructionsINTELInstBase
+    : public SPIRVInstTemplateBase {
+protected:
+  std::optional<ExtensionID> getRequiredExtension() const override {
+    return ExtensionID::SPV_INTEL_joint_matrix;
+  }
+  SPIRVCapVec getRequiredCapability() const override {
+    return getVec(
+        internal::CapabilityCooperativeMatrixInvocationInstructionsINTEL);
+  }
+};
+
+#define _SPIRV_OP(x, ...)                                                      \
+  typedef SPIRVInstTemplate<                                                   \
+      SPIRVCooperativeMatrixInvocationInstructionsINTELInstBase,               \
+      internal::Op##x##INTEL, __VA_ARGS__>                                     \
+      SPIRV##x##INTEL;
+_SPIRV_OP(CooperativeMatrixApplyFunction, true, 5)
```

Comment:
```suggestion
_SPIRV_OP(CooperativeMatrixApplyFunction, true, 4, true)
```
I guess we won't have the case with no Arguments, but this thing is usually defined as per spec, which says the instruction has only 4 necessary operands.
Could you please also clarify, what `Argument` values mean by spec?

Reply:
The suggestion is applied by the next commit.

About arguments, by spec, "Argument N is the object to copy to parameter N".

Reply:
I have changed it. And it should be:
_SPIRV_OP(CooperativeMatrixApplyFunction, true, 5)
the spec changes will be the following:
before
<html>
<body>
<!--StartFragment-->
4 + variable | 6448 | <id>Result Type | Result <id> | <id>Function | <id>Matrix | <id>, <id>, …​, <id>Argument 1, Argument 2, …​, Argument N
-- | -- | -- | -- | -- | -- | --


<!--EndFragment-->
</body>
</html>
after:
<html>
<body>
<!--StartFragment-->
5 | 6448 | <id>Result Type | Result <id> | <id>Function Object | <id>Matrix 
-- | -- | -- | -- | -- | -- | --

where Function Object is of a type pointer and should represent a function object. Please also note, that '4 + variable' that currently specified in the spec is a spec bug, it's actually '5 + variable'

<!--EndFragment-->
</body>
</html>

Reply:
Sorry for the confusion